var/spack/repos/builtin/packages/python/package.py

```diff
@@ -217,6 +217,10 @@ class Python(Package):
     # https://github.com/python/cpython/pull/16717
     patch("intel-3.7.patch", when="@3.7.1:3.7.5 %intel")
 
+    # Fix curses/readline detection logic to not be triggered by path name
+    # https://github.com/spack/spack/issues/34872
+    patch("curses.patch", when="@:3.11")
```

Comment:
Evidently I hadn't had enough coffee when I first read your comment. Anyhow, spack printed `==> Ran patch() for python` for every version without any error messages.
As I noted below, I'm fairly certain that the patch would break for versions released in 2010 and earlier (so roughly `@:2.8`). Is it necessary to check for the exact lower bound?

Reply:
Nope, don't care about versions older than 3.7